lib/orcid/pub_author_mapper.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+module Orcid
+  # Maps from pub_hash author to Orcid Contributor.
+  class PubAuthorMapper
+    # Maps to Orcid Contributor.
+    # @param [Hash] author_hash author
+    # @return [Hash] contributor
+    def self.map(author_hash)
+      new(author_hash).map
+    end
+
+    def initialize(author_hash)
+      @author_hash = author_hash
+    end
+
+    def map
+      {
+        "contributor-orcid": nil,
+        "credit-name": {
+          value: map_credit_name
+        },
+        "contributor-email": nil,
+        "contributor-attributes": {
+          "contributor-sequence": nil,
+          "contributor-role": author_hash[:role].presence || 'author'
+        }
+      }
+    end
+
+    private
+
+    attr_reader :author_hash
+
+    def map_credit_name
+      clean_name(author_hash[:name].presence) ||
```

Comment:
We take various strategies to constructing a name. The ordering of those strategies is deliberate.